### ISSUE TITLE
[Crash] CFRelease() called with NULL

### ISSUE THREAD

**Issue by IvanChan** [rs/SDWebImage#1968]
ver: 4.0.0

```
Application Specific Information:
*** CFRelease() called with NULL ***


Thread 32 name:  Dispatch queue: NSOperationQueue 0x11c7c6f70 :: NSOperation 0x111571670 (QOS: DEFAULT)
Thread 32 Crashed:
0   CoreFoundation                	0x0000000190da7290 CFRelease + 120
1   MyAPP                         	0x000000010015b2c8 -[SDWebImageDownloaderOperation URLSession:dataTask:didReceiveData:] (SDWebImageDownloaderOperation.m:379)
2   MyAPP                         	0x0000000100158e38 -[SDWebImageDownloader URLSession:dataTask:didReceiveData:] (SDWebImageDownloader.m:281)
3   CFNetwork                     	0x00000001916a78d0 __67-[NSURLSession delegate_dataTask:didReceiveData:completionHandler:]_block_invoke.242 + 36
4   Foundation                    	0x000000019199d804 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
5   Foundation                    	0x00000001918e2760 -[NSBlockOperation main] + 96
6   Foundation                    	0x00000001918d2b18 -[__NSOperationInternal _start:] + 612
7   Foundation                    	0x000000019199fba0 __NSOQSchedule_f + 228
8   libdispatch.dylib             	0x000000018fd869a0 _dispatch_client_callout + 16
9   libdispatch.dylib             	0x000000018fd94ad4 _dispatch_queue_serial_drain + 928
10  libdispatch.dylib             	0x000000018fd8a2cc _dispatch_queue_invoke + 884
11  libdispatch.dylib             	0x000000018fd96a50 _dispatch_root_queue_drain + 540
12  libdispatch.dylib             	0x000000018fd967d0 _dispatch_worker_thread3 + 124
13  libsystem_pthread.dylib       	0x000000018ff8f1d0 _pthread_wqthread + 1096
14  libsystem_pthread.dylib       	0x000000018ff8ed7c start_wqthread + 4
```

Somehow, imageSource can be nil
```
SDWebImageDownloaderOperation.m

        // Update the data source, we must pass ALL the data, not just the new bytes
        CGImageSourceRef imageSource = CGImageSourceCreateWithData((__bridge CFDataRef)self.imageData, NULL);

................

        CFRelease(imageSource);  

```

**Comment by bpoplauschi** [comment]
@IvanChan please provide all the details that are required in the ISSUE Template.

**Comment by dreampiggy** [comment]
I create an PR to fix this issue: #1978

**Comment by bpoplauschi** [comment]
Should be fixed by #1978. Thanks @dreampiggy
